target/linux/generic/files/drivers/net/phy/rtl8366_smi.c

```diff
@@ -919,6 +969,21 @@ static int rtl8366_smi_mii_init(struct rtl8366_smi *smi)
 {
 	int ret;
 	int i;
+	struct mii_bus *bus;
+	struct device_node *np;
+
+	if (smi->mdio_enabled) {
+		np = of_parse_phandle(smi->parent->of_node, "mdio", 0);
+
+		if (np) {
+			bus = of_mdio_find_bus(np);
+
+			if (bus) {
+				smi->mii_bus = bus;
+				return 0;
+			} else return -ENODEV;
```

Comment:
I suppose that driver set for rtl8366/7 should be rewritten so common functions will be brought out to function library rather additional driver (rtl8366_smi). No other driver than mentioned uses that driver. What do you think?